crates/avian_derive/src/lib.rs

```diff
@@ -2,19 +2,51 @@
 
 use proc_macro::TokenStream;
 
-use quote::{quote, quote_spanned};
+use proc_macro_error::{abort, emit_error, proc_macro_error};
+use quote::quote;
 use syn::{parse_macro_input, spanned::Spanned, Data, DeriveInput};
 
-// Modified macro From the discontinued Heron,
-// see https://github.com/jcornaz/heron/blob/main/macros/src/lib.rs
+// Modified macro from the discontinued Heron
+// https://github.com/jcornaz/heron/blob/main/macros/src/lib.rs
+/// A derive macro for defining physics layers using an enum.
+///
+/// Each variant of the enum represents a layer. Each layer has a unique bit determined by
+/// the order of the variants. The bit value can be retrieved using the `to_bits` method.
+///
+/// # Requirements
+///
+/// - The enum must have at most 32 variants.
+/// - The enum variants must not have any fields.
+/// - The enum must have a default variant with the `#[default]` attribute.
+///   - The first bit `1 << 0` will *always* be reserved for the default layer.
+///     The bit values of the other layers are determined by their order in the enum, starting from `1 << 1`.
+///
+/// # Example
+///
+/// ```ignore
+/// #[derive(PhysicsLayer, Default)]
```

Comment:
```suggestion
/// #[derive(PhysicsLayer, Clone, Copy, Debug, Default)]
```
to keep parity with the test you just added.